### PR TITLE
[BUG][Discover] Enable 'Back to Top' Feature in Discover for scrolling to top

### PR DESCRIPTION
**Pull request by ananzh**
### Description
dscCanvas is the one with scrollable prop. Set window.scrollTo(0, 0) on table will not work. In this PR, we add a ref to EuiPanel directly.

### Issues Resolved
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6006

## Screenshot

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/79961084/12b38b56-e867-4a08-9a58-52c814a19320

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by ananzh** [comment]
Add Function Test Suite 11: Infinity Scroll in Legacy Table in https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5713 to function meta issue.
We will add a new test suite in the functional test repo to test this bug.

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,7 +17,8 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - [BUG][Discover] Allow saved sort from search embeddable to load in Dashboard ([#5934](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5934))
 - [BUG][Discover] Add key to index pattern options for support deplicate index pattern names([#5946](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5946))
 - [Discover] Fix table cell content overflowing in Safari ([#5948](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5948))
-- [BUG][MD]Fix schema for test connection to separate validation based on auth type([#5997](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5997))
+- [BUG][MD]Fix schema for test connection to separate validation based on auth type ([#5997](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5997))
+- [BUG][Discover] Enable 'Back to Top' Feature in Discover for scrolling to top ([#6008](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6008))
```
Comment (AMoo-Miki) [inline CHANGELOG.md:21]:
```suggestion
- [Discover] Enable 'Back to Top' Feature in Discover for scrolling to top ([#6008](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6008))
```

**Review by AMoo-Miki — APPROVED** [review]

**Review by ashwin-pc — APPROVED** [review]
Ideally I'd pass the panel ref down the tree instead of the function itself since a ref will not cause any rerenders and can be used for other purposes as well. If you can change the prop being sent down to that, let's do it.

**Review by Flyingliuhub — APPROVED** [review]

**Comment by ananzh** [comment]
> Ideally I'd pass the panel ref down the tree instead of the function itself since a ref will not cause any rerenders and can be used for other purposes as well. If you can change the prop being sent down to that, let's do it.

k. let me try it.

**Pull request merged**